cmd/derper/derper.go

```diff
@@ -146,6 +148,21 @@ func main() {
 		log.Fatalf("invalid server address: %v", err)
 	}
 
+	if *runSTUN {
+		if *stunSubprocess {
+			if *stunOnly {
+				log.SetPrefix(fmt.Sprintf("stun(%d) ", os.Getpid()))
+				log.Printf("Starting in stun-only mode.")
+				go printSTUNStats(ctx, os.Stdout, time.Second*10)
```

Comment:
May I suggest to consider having the subprocess expose its own debug http server? This will improve our ability to introspect it via pprof, and will allow adding them separately to Prometheus. This will give us a bunch of built-in Go runtime metrics for free (gc metrics, goroutine count, memstats, cpu time) and help avoid bespoke metric serializaton machinery.

Reply:
That change ends up being a lot bigger once it is rationalized, because the basic take on process supervision and argument delegation taken here no longer makes so much sense. At that point we may as well split out a stund entirely, and manage it separately - but doing so then also requires new deployment machinery as well. I understand the motivations you're describing, it just may block us addressing this issue again, which has already been stuck for months due to time-cost trade-offs - e.g. see the PR that this supersedes.

Reply:
I guess it's fixit-friday so I did it anyway: #10757